straxen/rundb.py

```diff
@@ -136,16 +134,12 @@ def _data_query(self, key):
             'data': {
                 '$elemMatch': {
                     'type': key.data_type,
-                    # TODO remove the meta.lineage since this doc
-                    #  entry is deprecated.
-                    '$and': [{'$or': [
-                        {'meta.lineage': key.lineage},
+                    '$and': [
```

Comment:
If we include the lineage hash and the data type in the top level of the data array, then I think we make this query a bit more clear: 

```
{'data': {'$elemMatch': {'type': key.data_type, 'lineage_hash': key.lineage_hash, 'location': 'UC_DALI_USERDISK', 'status': 'transferred'}}}
```

I see this is part of the `available_query` -- not sure exactly best way to implement.

Reply:
Right, that is at least the $or field which is not per se needed here but we could change:
```
{'did': {'$regex': f'/*{key.data_type}-{key.lineage_hash}'}}
```
to
```
'type': key.data_type, 'lineage_hash': key.lineage_hash
```